### DISCUSSION TITLE
If hast is deprecated, why types are there?

### DISCUSSION THREAD

**Discussion by felixsanz** [rehypejs/rehype#105] [General] [answered]
Example: https://github.com/rehypejs/rehype-remark/blob/main/lib/index.js#L3

The `hast` package is deprecated, but the types are imported from that package, why?

**Comment by ChristianMurphy** [comment, marked as answer]
There is a distinction between `hast` the implementation, which now is known as rehype https://github.com/rehypejs/rehype
And hast the standard for AST nodes https://github.com/syntax-tree/hast

The implementation, formerly hosted at https://www.npmjs.com/package/hast is deprecated, only https://www.npmjs.com/package/rehype should be used.
https://www.npmjs.com/package/@types/hast is a types only package, which represents the standard https://github.com/syntax-tree/hast in TypeScript notation, and is leveraged by all `hast-util-*` and `rehype-*` packages as part of their typings.

**Reply by felixsanz** [reply]
thank you for clearing it out so fast!
